File: files/AzureOpenAIPoc/CSharp/PoC.cs
using System;
using System.Net;
using System.IO;
using System.Collections.Generic;
using System.Text.Json;
using System.Text.Json.Serialization;
using System.Threading.Tasks;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.Configuration.Json;
using Azure;

// Add Azure OpenAI package (Add code here)


namespace OpenAI_Chat
{  
    class Program  
    {
        public static void Main()  
        {  
            try  
            {  
                Utils.InitLog();
  
                IConfiguration config = new ConfigurationBuilder()
                    .AddJsonFile("appsettings.json")
                    .Build();
                string oaiEndpoint = config["AzureOAIEndpoint"] ?? "";
                string oaiKey = config["AzureOAIKey"] ?? "";
                string oaiModelName = config["AzureOAIModelName"] ?? "";

                // Initialize the Azure OpenAI client (Add code here)
                // This line should look like:
                //      CORRECT_TYPE client = new CORRECT_TYPE( ... );
                
                

                var functions = new Dictionary<int, Action<OpenAIClient, string>> {
                    { 1, function1 }, 
                    { 2, function2 }, 
                    { 3, function3 }, 
                    { 4, function4 }  
                };

                while (true) {
                    Console.WriteLine("1: Validate PoC\n" +
                        "2: Company chatbot\n" +
                        "3: Developer tasks\n" +
                        "4: Use company data\n" +
                        "\'quit\' to exit the program\n");
                    string userInput = (Console.ReadLine() ?? "").Trim().ToLower();                    
                    
                    if (userInput == "quit") {
                        break;
                    }
                    
                    int inputKey = int.Parse(userInput);

                    if (functions.ContainsKey(inputKey)) {
                        functions[inputKey](client, oaiModelName);
                    }
                    else {
                        Console.WriteLine("Invalid input. Please enter number 1,2,3, or 4.");
                    }
                } 
            }  
            catch (Exception ex)  
            {  
                Console.WriteLine(ex);  
            }  
        }

        // Task 1: Validate PoC
        static void function1(OpenAIClient client, string oaiModelName) {
            var promptInput = Utils.GetPromptInput("Task 1: Validate PoC\n", "sample-text.txt"); 
  
            // Build completion options object (Add code here)
            // This line should look like:
            //      CORRECT_TYPE options = new CORRECT_TYPE() { ... };
            var options =            

            Utils.WriteLog("API Parameters: ", options);

            // Send request to Azure OpenAI model (Add code here)
            // This line should look like:
            //      CORRECT_TYPE response = client.CORRECT_METHOD(...);
            var response =             

            Utils.WriteLog("Response:\n", response);  
            Console.WriteLine("Response: " + response.Choices[0].Message.Content + "\n"); 
        }

        // Task 2: Company chatbot
        static void function2(OpenAIClient client, string oaiModelName) {
            /* Uncomment this function when you're ready to work on Task 2
            var promptInput = Utils.GetPromptInput("Task 2: Company chatbot\n", "sample-text.txt"); 
  
            // Build completion options object (Add code here)
            // This line should look like:
            //      CORRECT_TYPE options = new CORRECT_TYPE() { ... };
            var options = 

            Utils.WriteLog("API Parameters: ", options);

            // Send request to Azure OpenAI model (Add code here)
            // This line should look like:
            //      CORRECT_TYPE response = client.CORRECT_METHOD(...);
            var response = 

            Utils.WriteLog("Response:\n", response);  
            Console.WriteLine("Response: " + response.Choices[0].Message.Content + "\n");
            */
        }

        // Task 3: Developer tasks
        static void function3(OpenAIClient client, string oaiModelName) {
            /* Uncomment this function when you're ready to work on Task 3
            var promptInput = Utils.GetPromptInput("Task 3: Developer tasks\n", "sample-text.txt");  
  
            // Build completion options object (Add code here)
            // This line should look like:
            //      CORRECT_TYPE options = new CORRECT_TYPE() { ... };
            var options = 

            Utils.WriteLog("API Parameters: ", options);

            // Send request to Azure OpenAI model (Add code here)
            // This line should look like:
            //      CORRECT_TYPE response = client.CORRECT_METHOD(...);
            var response = client.GetChatCompletions(options);

            Utils.WriteLog("Response:\n", response);  
            Console.WriteLine("Response: " + response.Choices[0].Message.Content + "\n");  
            */
        }

        // Task 4: Use company data
        static void function4(OpenAIClient client, string oaiModelName) {
            /* Uncomment this function when you're ready to work on Task 4
            string promptInput = Utils.GetPromptInput("Task 4: Use company data\n", "sample-text.txt");

            // Build completion options object (Add code here)
            // This section should include something like:
            //      CORRECT_TYPE options = new CORRECT_TYPE() { ... };
            var options = 

            Utils.WriteLog("API Parameters: ", options);

            // Send request to Azure OpenAI model (Add code here)
            // This line should look like:
            //      CORRECT_TYPE response = client.CORRECT_METHOD(...);
            var response =            
            
            Utils.WriteLog("Response:\n", response);  
            Console.WriteLine("Response: " + response.Choices[0].Message.Content + "\n"); 
            */
        }     
    }  
}  

File: files/AzureOpenAIPoc/Python/PoC.py
import os
from dotenv import load_dotenv
import utils
# Add OpenAI import. (Add code here)
from openai import AzureOpenAI


def main(): 
        
    try:     
        load_dotenv()
        utils.initLogFile()
        azure_oai_endpoint = os.getenv("AZURE_OAI_ENDPOINT")
        azure_oai_key = os.getenv("AZURE_OAI_KEY")
        azure_oai_model = os.getenv("AZURE_OAI_MODEL")
        
        # Define Azure OpenAI client (Add code here)
        client=AzureOpenAI(
            azure_endpoint=azure_oai_endpoint,
            api_key=azure_oai_key,
            api_version="0613"
        )
        

        function_map = {
            "1": function1,
            "2": function2,
            "3": function3,
            "4": function4
        }

        while True:
            print('1: Validate PoC\n' +
                  '2: Company chatbot\n' +
                  '3: Developer tasks\n' +
                  '4: Use company data\n' +
                  '\'quit\' to exit the program\n')
            command = input('Enter a number:')
            if command.strip() in function_map:
                function_map[command](client, azure_oai_model)
            elif command.strip().lower() == 'quit':
                print('Exiting program...')
                break
            else :
                print("Invalid input. Please enter number 1, 2, 3, 4, or 5.")

    except Exception as ex:
        print(ex)

# Task 1: Validate PoC
def function1(aiClient, aiModel):
    inputText = utils.getPromptInput("Task 1: Validate PoC", "sample-text.txt")
    
    # Build messages to send to Azure OpenAI model. (Add code here)
    messages=[
            {"role": "", "": inputText}
        ]
    

    # Define argument list (Add code here)
    apiParams = {
        "messages": messages,
    }
    

    utils.writeLog("API Parameters:\n", apiParams)

    # Call chat completion connection. (Add code here)
    # Use the call name and **apiParams to reference our argument list
    response = ""
    

    utils.writeLog("Response:\n", str(response))
    print("Response: " + response.choices[0].message.content + "\n")
    return response

# Task 2: Company chatbot
def function2(aiClient, aiModel):
    inputText = utils.getPromptInput("Task 2: Company chatbot", "sample-text.txt")
    
    # Build messages to send to Azure OpenAI model. (Add code here)
    messages=[
            {"role": "", "": inputText}
        ]
    

    # Define argument list (Add code here)
    apiParams = {
        "messages": messages,
    }
    

    utils.writeLog("API Parameters:\n", apiParams)

    # Call chat completion connection. (Add code here)
    # Use the call name and **apiParams to reference our argument list
    response = ""
    

    utils.writeLog("Response:\n", str(response))
    print("Response: " + response.choices[0].message.content + "\n")
    return response

# Task 3: Developer tasks
def function3(aiClient, aiModel):
    inputText = utils.getPromptInput("Task 3: Developer tasks", "sample-text.txt")
    
    # Build messages to send to Azure OpenAI model. (Add code here)
    messages=[
            {"role": "", "": inputText}
        ]
    

    # Define argument list (Add code here)
    apiParams = {
        "messages": messages,
    }
    
    utils.writeLog("API Parameters:\n", apiParams)

    # Call chat completion connection. (Add code here)
    # Use the call name and **apiParams to reference our argument list
    response = ""

    
    utils.writeLog("Response:\n", str(response))
    print("Response: " + response.choices[0].message.content + "\n")
    return response 

# Task 4: Use company data
def function4(aiClient, aiModel):
    inputText = utils.getPromptInput("Task 4: Use company data", "sample-text.txt")
    
    # Build messages to send to Azure OpenAI model. (Add code here)
    messages=[
            {"role": "", "": inputText}
        ]
    

    # Define connection and argument list (Add code here)
    apiParams = {
        "messages": messages,
    }
    

    utils.writeLog("API Parameters:\n", apiParams)

    # Call chat completion connection. Will be the same as function1 (Add code here)
    # Use the call name and **apiParams to reference our argument list
    response = ""
    

    utils.writeLog("Response:\n", str(response))
    print("Response: " + response.choices[0].message.content + "\n")
    return

# Call main function. Do not modify.
if __name__ == '__main__': 
    main()
